lib/sassc/rails/template.rb

```diff
@@ -34,13 +34,36 @@ def call(input)
         }
       }.merge!(config_options) { |key, left, right| safe_merge(key, left, right) }
 
+      if Rails.application.config.assets.debug && Sprockets::VERSION.start_with?("4")
+        options.merge!(
+          source_map_contents: false,
+          source_map_file: "#{input[:filename]}.map",
+          omit_source_map_url: true,
+        )
+      end
+
       engine = ::SassC::Engine.new(input[:data], options)
 
       css = Sprockets::Utils.module_include(::SassC::Script::Functions, @functions) do
         engine.render
       end
 
-      context.metadata.merge(data: css)
+      if Rails.application.config.assets.debug && Sprockets::VERSION.start_with?("4")
+        begin
+          map = Sprockets::SourceMapUtils.format_source_map(JSON.parse(engine.source_map), input)
+          map = Sprockets::SourceMapUtils.combine_source_maps(input[:metadata][:map], map)
+
+          engine.dependencies.each do |dependency|
+            context.metadata[:dependencies] << Sprockets::URIUtils.build_file_digest_uri(dependency.filename)
+          end
+        rescue SassC::NotRenderedError
+          map = input[:metadata][:map]
+        end
+
+        context.metadata.merge(data: css, map: map)
```

Comment:
`map` metadata wasn't transmitted to sprockets and wasn't aware of dependencies, so it wasn't able to generate the source map with complete data.